ares/n64/cpu/interpreter-ipu.cpp

```diff
@@ -138,7 +139,13 @@ auto CPU::CACHE(u8 operation, cr64& rs, s16 imm) -> void {
 
   case 0x10: {  //icache hit invalidate
     auto& line = icache.line(address);
-    if(line.hit(address)) line.valid = 0;
+    // Invalidate recompiler also on non-hits. icache invalidate is a signal
+    // that something was changed in memory, even though the memory is not in
+    // the icache anymore.
+    cpu.recompiler.invalidate(address);
```

Comment:
Consider moving this recompiler invalidation block to the 'miss; case in https://github.com/ares-emulator/ares/blob/dc5ff13b5ca61e61a4057c0a4de143440181909d/ares/n64/cpu/icache.cpp#L47

This would ensure that the host recompiler cache is cleared when the n64 icache suffers a cache miss; so it would catch scenarios that just using the cache opcode would not.

Reply:
turns out I may have been incorrect here; The step function is called after the opcode is implemented, so it's too late to invalidate